File: library/src/main/java/com/jw/library/utils/RotateTransformation.java
package com.jw.library.utils;

import android.content.Context;
import android.graphics.Bitmap;

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;

public class RotateTransformation extends BitmapTransformation {

    private int rotateRotationAngle = 0;

    public RotateTransformation(Context context, int rotateRotationAngle) {
        super(context);

        this.rotateRotationAngle = rotateRotationAngle;
    }

    @Override
    protected Bitmap transform(BitmapPool pool, Bitmap toTransform, int outWidth, int outHeight) {
        return BitmapUtil.rotateBitmapByDegree(toTransform, rotateRotationAngle);
    }

    @Override
    public String getId() {
        return "rotate" + rotateRotationAngle;
    }
}

File: library/src/main/java/com/jw/library/utils/RotateTransformation.kt
package com.jw.library.utils

import android.content.Context
import android.graphics.Bitmap

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
import java.security.MessageDigest

class RotateTransformation(context: Context, private val rotateRotationAngle: Int) :
    BitmapTransformation(context) {
    override fun updateDiskCacheKey(messageDigest: MessageDigest?) {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun transform(
        pool: BitmapPool,
        toTransform: Bitmap,
        outWidth: Int,
        outHeight: Int
    ): Bitmap {
        return BitmapUtil.rotateBitmapByDegree(toTransform, rotateRotationAngle)
    }

}